### ISSUE TITLE
[SEGV : traci-client.cc:431]

### ISSUE THREAD

**Issue by qri-hub** [vodafone-chair/ns3-sumo-coupling#2]
I did some tuning of ns3-coupling-simple example. 

I deleted the NS_FATAL_ERROR on line 112 inside the callback function for node creation : 

nodeCounter; // increment count 
`std::function<Ptr<Node> ()> setupNewWifiNode = [&] () -> Ptr<Node>
  {
  	if (nodeCounter >= nodePool.GetN())
	    //NS_FATAL_ERROR("Node Pool empty!: " << nodeCounter << " nodes created.");
	    NS_LOG_INFO("Node Pool Empty");
	else{
	    Ptr<Node> includedNode = nodePool.Get(nodeCounter);
            NS_LOG_INFO("++++ nid: "<< includedNode->GetId());
            ++nodeCounter; // increment count
}
`
Thus it passes the function and SynchroniseVehicleNodeMap is called. This is where the program crashes:
`TraciClient::SynchroniseVehicleNodeMap()
  {
    NS_LOG_FUNCTION(this);

    try
      {
        // get departed and arrived sumo vehicles since last simulation step
        std::vector<std::string> sumoVehicles;
        GetSumoVehicles(sumoVehicles);

        // iterate over all sumo vehicles with changes; include departed vehicles, exclude arrived vehicles
        for (std::vector<std::string>::iterator it = sumoVehicles.begin(); it != sumoVehicles.end(); ++it)
          {
            // get current vehicle
            std::string veh(*it);

            // search for vehicle in vehicleNodeMap
            std::map<std::string, Ptr<Node> >::iterator pos = m_vehicleNodeMap.find(veh);

            // if it is already in the map, remove it and exclude node

              if (pos != m_vehicleNodeMap.end())
                {
                  // get corresponding ns3 node
                  Ptr<ns3::Node> exNode = m_vehicleNodeMap.at(veh);

                  // call exclude function for this node
                  m_excludeNode(exNode);

                  // unregister in map
                  m_vehicleNodeMap.erase(veh);
                }
              else // if it is not in the map, create a new ns3 node for it
                {
                  // create new node by calling the include function
                  Ptr<ns3::Node> inNode = m_includeNode();

                  // register in the map (link vehicle to node!)
                  m_vehicleNodeMap.insert(std::pair<std::string, Ptr<Node>>(veh, inNode)); //SEGV occurs here
                }
          }
      }
    catch (std::exception& e)
      {
        NS_FATAL_ERROR("SUMO was closed unexpectedly while updating the vehicle node map: " << e.what());
      }
  }
`

I think that the problem seems to be that the number of sumo vehicles gets greater than the Nodepool
therefore it overflows the Map.

Do you have an idea on how i could prevent further nodes being allocated to the vehicles created by sumo. 
In short : 
How can I tell the Ns3 to build nodes for the first n Sumo-vehicles (and ignore the followings) ?

One more thing, 
The crash happens while there is only 52 vehicles inside sumo (although i created 100 nodes).
I think it is because the nodes allowed to sumo vehicles that are deleted are not available for reallocation.
Therefore is it possible to reallocate these nodes on the fly ( so that a single node could be used by another vehicle once the first is deleted from the simulation) ?

**Comment by qri-hub** [comment]
Solved by reusing nodes from the pool;
